code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -4352,6 +4339,32 @@
 				available_snacks = shuffle(available_snacks)
 			counter++
 
+/obj/item/weapon/reagent_containers/food/snacks/pie/nofruitpie/proc/getnofruit(mob/user, obj/item/weapon/W = null)
+	if(!switching || !current_path)
+		return
+	verbs -= /obj/item/weapon/reagent_containers/food/snacks/pie/nofruitpie/verb/pick_leaf
+	switching = 0
+	var/N = rand(1,3)
+	if(get_turf(user))
+		switch(N)
+			if(1)
+				playsound(user, 'sound/weapons/genhit1.ogg', 50, 1)
+			if(2)
+				playsound(user, 'sound/weapons/genhit2.ogg', 50, 1)
+			if(3)
+				playsound(user, 'sound/weapons/genhit3.ogg', 50, 1)
```

Comment:
Can't you just 
``playsound(user, 'sound/weapons/genhit[rand (1,3)].ogg', 50, 1)``

Reply:
I'm pretty sure BYOND doesn't allow the use of `[]` in filenames.